ppl/cmd/pgctl/migrate/command.go

```diff
@@ -0,0 +1,79 @@
+package testdb
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/ppl/cmd/pgctl/root"
+	"github.com/brimsec/zq/ppl/zqd/db/postgresdb"
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"github.com/mccanne/charm"
+)
+
+var Migrate = &charm.Spec{
+	Name:  "migrate",
+	Usage: "migrate [-m migration dir] [postgres options]",
+	Short: "perform a schema migration on a database",
+	New:   New,
+}
+
+type Command struct {
+	*root.Command
+	postgres   postgresdb.Config
+	migrations string
+}
+
+func init() {
+	root.CLI.Add(Migrate)
+}
+
+func New(parent charm.Command, fs *flag.FlagSet) (charm.Command, error) {
+	c := &Command{Command: parent.(*root.Command)}
+	fs.StringVar(&c.migrations, "m", "", "migrations directory")
+	c.postgres.SetFlags(fs)
+	return c, nil
+}
+
+func (c *Command) Run(args []string) error {
+	if err := c.Init(&c.postgres); err != nil {
+		return err
+	}
+	if c.migrations == "" {
+		return errors.New("argument '-m' (migrations directory) is required")
+	}
+
+	u, err := iosrc.ParseURI(c.migrations)
+	if err != nil {
+		return err
+	}
+
+	m, err := migrate.New(u.String(), c.postgres.String())
```

Comment:
This is a high-level question. I'm including it here because this code, which all looks good, is the context for the question. 

Why do we have our own CLI for migrations (that uses the "New" call) rather that using the CLI implemented by the golang-migrate project?

E.g. migrate -source file://path/to/migrations -database postgres://localhost:5432/database up 2

Reply:
We could do that, I was just thinking that since we already need pgctl to create/destroy test database instances it wouldn't be that difficult to just make another command vs having to wire up another binary into our build system. Having our commands does have some niceties- for instance we can use the same cli options for specifying the postgres connection details that we use in zqd listen. We might also want to include more custom logging, etc.